config/routes.rb

```diff
@@ -1,5 +1,14 @@
 Rails.application.routes.draw do
   devise_for :users
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
- root 'homepage#index'
+  resources :clients
+  root 'homepage#index'
+  namespace :admin do
+    root to:'admin#index'
+    resources :admin do
+      member do
+        get 'change_status'
+        get 'promote_to_manager'
```

Comment:
i think i mentioned before, this updates the role, so use update_role as action name.
it not only promotes but demotes as well.

Reply:
Have already changed it with Update_role.